velox/CMakeLists.txt

```diff
@@ -11,31 +11,57 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-add_subdirectory(aggregates)
+
+# VELOX_BASE
 add_subdirectory(buffer)
 add_subdirectory(common)
-add_subdirectory(connectors)
 add_subdirectory(core)
-add_subdirectory(dwio)
-add_subdirectory(examples)
-add_subdirectory(exec)
-add_subdirectory(expression)
-add_subdirectory(external/date)
-add_subdirectory(external/md5)
-add_subdirectory(functions)
-add_subdirectory(parse)
 add_subdirectory(serializers)
 add_subdirectory(type)
 add_subdirectory(vector)
 add_subdirectory(row)
 add_subdirectory(flag_definitions)
+add_subdirectory(external/date)
+add_subdirectory(external/md5)
+#
+
+# examples depend on expression
+if((${VELOX_ENABLE_EXAMPLES} AND ${VELOX_ENABLE_EXPRESSION})
+   OR ${VELOX_BUILD_TESTING})
+  add_subdirectory(examples)
+endif()
+
+if(${VELOX_ENABLE_EXPRESSION} OR ${VELOX_BUILD_TESTING})
```

Comment:
why the OR build testing? Because is that flag is on we want to build expression unit tests?

Reply:
> why the OR build testing? Because is that flag is on we want to build expression unit tests?

The logic here is you have to disable both VELOX_ENABLE_EXPRESSION and VELOX_BUILD_TESTING to really disable the component. if you enable VELOX_BUILD_TESTING, all components must be built. As we discussed earlier that's because tests in each component has deps beyond it's belonging component. On the other hand, if you just enable VELOX_ENABLE_EXPRESSION, the component will be built no matter you want to build tests or not.

Reply:
If so, would it be cleaner to handle this in the top-level cmake file, by overwriting all flags to ON if build testing in ON? Otherwise we'll have to add the OR every single time we use these variables.